cloud-config/deploy-run-scripts/es-install.sh

```diff
@@ -4,51 +4,79 @@
-}
-
-function append_with_user {
-  line="$1"
-  user="$2"
-  dest="$3"
-  echo "$line" | sudo -u $user tee -a $dest
-}
-
-
-# jvm options
-jvm_opts_filename='jvm.options'
-jvm_xms='-Xms'"$JVM_GIGS"'g'
-jvm_xmx='-Xmx'"$JVM_GIGS"'g'
-append_with_user "$jvm_xms" ubuntu "$opts_src/$jvm_opts_filename"
-append_with_user "$jvm_xmx" ubuntu "$opts_src/$jvm_opts_filename"
-copy_with_permission "$opts_src/$jvm_opts_filename" "$opts_dest/$jvm_opts_filename"
-
-# elasticsearch options
-es_opts_filename="$ES_OPT_FILENAME"
-if [ "$CLUSTER_NAME" == 'NONE' ]; then
-    echo 'Not a elasticsearch cluster'
-else
-    # Only append a cluster name if it is not 'NONE'
-    # like single demos do not have cluster names
-    cluster_name="cluster.name: $CLUSTER_NAME"
-    append_with_user "$cluster_name" ubuntu "$opts_src/$es_opts_filename"
+# Install es and add to boot service
+if [ ! "$ONLY_RESTART_ES" == 'yes' ]; then
+    opts_src='/home/ubuntu/encoded/cloud-config/deploy-run-scripts/conf-es'
+    opts_dest='/etc/elasticsearch'
+
+
+    function copy_with_permission {
+        src_file="$1"
+        dest_file="$2"
+        sudo -u root cp "$src_file" "$dest_file"
+        sudo -u root chown root:elasticsearch "$dest_file"
+    }
+
+    function append_with_user {
+      line="$1"
+      user="$2"
+      dest="$3"
+      echo "$line" | sudo -u $user tee -a $dest
+    }
+
+
+    # jvm options
+    jvm_opts_filename='jvm.options'
+    jvm_xms='-Xms'"$JVM_GIGS"'g'
+    jvm_xmx='-Xmx'"$JVM_GIGS"'g'
+    append_with_user "$jvm_xms" ubuntu "$opts_src/$jvm_opts_filename"
+    append_with_user "$jvm_xmx" ubuntu "$opts_src/$jvm_opts_filename"
+    copy_with_permission "$opts_src/$jvm_opts_filename" "$opts_dest/$jvm_opts_filename"
+
+    # elasticsearch options
+    es_opts_filename="$ES_OPT_FILENAME"
+    if [ "$CLUSTER_NAME" == 'NONE' ]; then
+        echo 'Not a elasticsearch cluster'
+    else
+        # Only append a cluster name if it is not 'NONE'
+        # like single demos do not have cluster names
+        cluster_name="cluster.name: $CLUSTER_NAME"
+        append_with_user "$cluster_name" ubuntu "$opts_src/$es_opts_filename"
+    fi
+    copy_with_permission "$opts_src/$es_opts_filename" "$opts_dest/elasticsearch.yml"
+
+    # Install discovery for clusters, maybe only needed for clusters
+    sudo /usr/share/elasticsearch/bin/elasticsearch-plugin install discovery-ec2
+    # Add es service and start
+    sudo /bin/systemctl enable elasticsearch.service
+    sudo systemctl start elasticsearch.service
 fi
-copy_with_permission "$opts_src/$es_opts_filename" "$opts_dest/elasticsearch.yml"
 
-# Setup/Restart
-update-rc.d elasticsearch defaults
-sudo /usr/share/elasticsearch/bin/elasticsearch-plugin install discovery-ec2
-service elasticsearch restart
+
+if [ -z "$CLUSTER_NAME" ]; then
+    # Only for single demos
+    # start es and wait for yellow or green
+    while true; do
+        es_status="$(curl -fsSL 'localhost:9201/_cat/health?h=status')"
```

Comment:
I don't think port 9201 should be hard coded here.

Reply:
that's a good idea.  We can pass in the port.  That parts isn't quite completed too.

Reply:
This was removed.  There was a bug where es wasn't starting but now he don't need to check if its's on.